akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala

```diff
@@ -99,6 +98,54 @@ object CoordinatedShutdown extends ExtensionId[CoordinatedShutdown] with Extensi
    */
   val PhaseActorSystemTerminate = "actor-system-terminate"
 
+  /**
+   * Reason for the shutdown, which can be used by tasks in case they need to do
+   * different things depending on what caused the shutdown. There are some
+   * predefined reasons, but external libraries applications may also define
+   * other reasons.
+   */
+  trait Reason
+
+  /**
+   * Scala API: The reason for the shutdown was unknown. Needed for backwards compatibility.
+   */
+  case object UnknownReason extends Reason
+
+  /**
+   * Java API: The reason for the shutdown was unknown. Needed for backwards compatibility.
+   */
+  def unknownReason: Reason = UnknownReason
+
+  /**
+   * Scala API: The shutdown was initiated by a JVM shutdown hook, e.g. triggered by SIGTERM.
+   */
+  object JvmExitReason extends Reason
```

Comment:
Just noticed some `Reason`'s are `case object` and some are `object` (see `UnknownReason` vs `JvmExitReason`) . Is there a rule for that? Wouldn't it be safe if all where `case object`?

Reply:
that was an oversight, they should all be `case object` so that they have a better `toString`, could you PR?